servers/rendering/dummy/storage/material_storage.cpp

```diff
@@ -36,7 +36,7 @@ MaterialStorage *MaterialStorage::singleton = nullptr;
 
 MaterialStorage::MaterialStorage() {
 	singleton = this;
-	ShaderCompiler::DefaultIdentifierActions actions;
+	auto actions = ShaderCompiler::DefaultIdentifierActions();
```

Comment:
We don't use `auto` aside from very specific cases (e.g. for some lambdas), so this should state the type explicitly.

Reply:
```c++
ShaderCompiler::DefaultIdentifierActions actions;
```

resulted in warnings about `actions` being uninitialized, so I updated this line; I had a feeling Godot doesn't use `auto`, but because the line got pretty long due to the edit I opted for using it. Would this be ok?

```c++
ShaderCompiler::DefaultIdentifierActions actions = ShaderCompiler::DefaultIdentifierActions();
```

Reply:
Well actually this seems to be outside the scope of this PR. There are many other occurrences of this code that would need to changed:
```
servers/rendering/shader_compiler.h:	struct DefaultIdentifierActions {
servers/rendering/shader_compiler.h:	String _dump_node_code(const ShaderLanguage::Node *p_node, int p_level, GeneratedCode &r_gen_code, IdentifierActions &p_actions, const DefaultIdentifierActions &p_default_actions, bool p_assigning, bool p_scope = true);
servers/rendering/shader_compiler.h:	DefaultIdentifierActions actions;
servers/rendering/shader_compiler.h:	void initialize(DefaultIdentifierActions p_actions);
servers/rendering/shader_compiler.cpp:String ShaderCompiler::_dump_node_code(const SL::Node *p_node, int p_level, GeneratedCode &r_gen_code, IdentifierActions &p_actions, const DefaultIdentifierActions &p_default_actions, bool p_assigning, bool p_use_scope) {
servers/rendering/shader_compiler.cpp:void ShaderCompiler::initialize(DefaultIdentifierActions p_actions) {
servers/rendering/renderer_rd/environment/sky.cpp:		ShaderCompiler::DefaultIdentifierActions actions;
servers/rendering/renderer_rd/environment/fog.cpp:		ShaderCompiler::DefaultIdentifierActions actions;
servers/rendering/renderer_rd/forward_mobile/scene_shader_forward_mobile.cpp:		ShaderCompiler::DefaultIdentifierActions actions;
servers/rendering/renderer_rd/storage_rd/particles_storage.cpp:		ShaderCompiler::DefaultIdentifierActions actions;
drivers/gles3/storage/material_storage.cpp:		ShaderCompiler::DefaultIdentifierActions actions;
drivers/gles3/storage/material_storage.cpp:		ShaderCompiler::DefaultIdentifierActions actions;
drivers/gles3/storage/material_storage.cpp:		ShaderCompiler::DefaultIdentifierActions actions;
drivers/gles3/storage/material_storage.cpp:		ShaderCompiler::DefaultIdentifierActions actions;
servers/rendering/renderer_rd/renderer_canvas_render_rd.cpp:		ShaderCompiler::DefaultIdentifierActions actions;
servers/rendering/renderer_rd/forward_clustered/scene_shader_forward_clustered.cpp:		ShaderCompiler::DefaultIdentifierActions actions;
servers/rendering/dummy/storage/material_storage.cpp:	ShaderCompiler::DefaultIdentifierActions actions;
```

There's two options, either all of these should be initialized with `= {};`, or the struct should be changed to fully initialize all its members (here `default_filter` and `default_repeat` would need to have a default value, the rest are fine already).

But that should be in a separate PR.

Reply:
Yes, these changes don't seem related to this Vulkan update at all.

Reply:
Submitted an issue regarding this problem:

- https://github.com/godotengine/godot/issues/88015